test/groovy/MavenExecuteTest.groovy

```diff
@@ -0,0 +1,41 @@
+import junit.framework.TestCase
+import org.junit.Before
+import org.junit.Test
+
+class MavenExecuteTest extends AbstractPiperUnitTest {
+
+    Map dockerParameters
+    List shellCalls
+
+    @Before
+    void setUp() {
+        super.setUp()
+
+        shellCalls = []
+        dockerParameters = [:]
+
+        helper.registerAllowedMethod("dockerExecute", [Map.class, Closure.class],
+            { parameters, closure ->
+                dockerParameters = parameters
+                closure()
+            })
+        helper.registerAllowedMethod('sh', [String], {s -> shellCalls.add(s)} )
+    }
+
+    @Test
+    void testExecuteBasicMavenCommand() throws Exception {
+        def script = loadScript("test/resources/pipelines/mavenExecuteTest/executeBasicMavenCommand.groovy",)
+            script.execute()
+            TestCase.assertEquals('maven:3.5-jdk-7', dockerParameters.dockerImage)
+            TestCase.assertTrue(shellCalls.contains('mvn clean install'))
```

Comment:
Test fails for me when running `mvn clean test`. Error message:
> `MavenExecuteTest.testExecuteBasicMavenCommand:29 expected:<maven:3.5-jdk-7> but was:<null>`

As a Unit Test it works, however.